torch/csrc/autograd/engine.h

```diff
@@ -56,10 +57,21 @@ struct GraphTask {
 
   struct ExecInfo {
     struct Capture {
+      Capture(const Capture&) = delete;
+      Capture(Capture&&) = default;
+
       Capture(int input_idx, int output_idx)
           : input_idx_(input_idx), output_idx_(output_idx) {}
       int input_idx_; // within Node inputs
       int output_idx_; // within the output vector of a GraphTask
+
+      // This hook will be executed before a grad is captured. The engine will
+      // capture the returned value.
+      struct GradCapturePreHook {
+        virtual ~GradCapturePreHook() = default;
+        virtual torch::Tensor operator()(const torch::Tensor& grad) = 0;
+      };
```

Comment:
nit: Maybe call this GradCaptureHook since we're actually calling the hook after the grad is captured in captured_vars_ now. Probably need to fix the comment as well?

Reply:
Will make the change.